### ISSUE TITLE
Some questions regarding model training and validation.

### ISSUE THREAD

**Issue by 3Dian1415926-xin** [cxliu0/PET#35]
Hi Author,

Thank you for the great work.When I was running your open-source code, I encountered a very strange issue. When I changed eval_freq=5 to eval_freq=1, without making any other modifications, I obtained two significantly different results. It seems that the frequency of validation also has an impact on the model's training. Even though the source code includes model.eval() during validation, have you (the author) ever encountered this issue before? If so, how did you resolve it?

As shown in the training logs below, the MAE obtained at the fifth epoch differs significantly between the two settings, and the loss values from epoch 2 to epoch 5 also show a noticeable difference.

I look forward to your reply.

# eval_freq=5
[ep0][lr 0.0001000][15.09s]{"train_loss": 0.621623741613852}
[ep1][lr 0.0001000][11.25s]{"train_loss": 0.4947314254335455}
[ep2][lr 0.0001000][11.20s]{"train_loss": **0.4989600503766859**}
[ep3][lr 0.0001000][10.73s]{"train_loss": 0.4701503176946898}
[ep4][lr 0.0001000][11.01s]{"train_loss": 0.4603341705090291}
[ep5][lr 0.0001000][11.08s]{"train_loss": 1.0641800503473025}

epoch:5, **_mae:225.65384615384616_**, mse:293.019248889523, time13.051184892654419,
best mae:225.65384615384616, best epoch: 5

# eval_freq=1
[ep 0][lr 0.0001000][13.38s]{ "train_loss": 0.621623736781043} 
[ep 1][lr 0.0001000][11.20s]{ "train_loss": 0.4947314222116728}
epoch:1, mae:338.0274725274725, mse:420.63321575667584, time11.947060823440552, 
best mae:338.0274725274725, best epoch: 1
####  Start testing
[ep 2][lr 0.0001000][10.94s]{ "train_loss": **0.4869693324372575**}
epoch:2, mae:247.96703296703296, mse:319.44367850415153, time11.986369371414185, 
best mae:247.96703296703296, best epoch: 2

[ep 3][lr 0.0001000][10.94s]{ "train_loss": 0.48861463891493306 }
epoch:3, mae:232.8021978021978, mse:287.44346792499437, time12.006171464920044, 
best mae:232.8021978021978, best epoch: 3

[ep 4][lr 0.0001000][11.24s]{ "train_loss": 0.4665500102816401}
epoch:4, mae:242.5, mse:297.0287013918848, time12.096357822418213, best mae:232.8021978021978, best epoch: 3

[ep 5][lr 0.0001000][11.09s]{ "train_loss": 1.0212259034852724}
epoch:5, **_mae:424.54395604395603_**, mse:634.3506370842791, time12.772287607192993, 
best mae:232.8021978021978, best epoch: 3

![Image](https://github.com/user-attachments/assets/014b1e65-bedb-4a22-aeb0-cb50b081559c)

**Comment by cxliu0** [comment]
Thank you for the question. I believe this is related to the mechanism of PyTorch, as evaluation may affect seed generation in the dataloader. As discussed in [this issue](https://github.com/pytorch/pytorch/issues/80119#issuecomment-1164630341), you may manually pass the generator when defining the training dataloader.

**Comment by 3Dian1415926-xin** [comment]
Hi Author,

Thank you so much for your quick and helpful reply! It really helped me understand why changing the evaluation frequency affected the training results. I really appreciate it.

I'm currently working on reproducing the JHU-CROWD++ dataset. One thing I'm not sure about is how to handle large images. The paper mentions that the maximum allowed size is 2048 on the longer side — do you usually resize them directly using cv2.resize, or crop them into smaller patches where each has a max side length of 2048?

Thanks again for your help!

Best regards

**Comment by cxliu0** [comment]
Hi, the JHU-CROWD++ dataset is preprocessed before training, where the longer size of each image is limited to 2048. You can refer to [preprocess_dataset.py](https://github.com/cxliu0/PET/blob/main/preprocess_dataset.py#L65) for more details.
